hydra-node/src/Hydra/Node/InputQueue.hs

```diff
@@ -62,4 +72,33 @@ createInputQueue = do
             n <- readTVar numThreads
             isEmpty' <- isEmptyTQueue q
             pure (isEmpty' && n == 0)
+      , done = \_ -> pure ()
+      }
+
+createPersistentInputQueue ::
+  ( MonadDelay m
+  , MonadIO m
+  , MonadSTM m
+  , MonadCatch m
+  , MonadFail m
+  , Eq e
+  , ToCBOR e
+  , FromCBOR e
+  ) =>
+  FilePath ->
+  Natural ->
+  m (InputQueue m e)
+createPersistentInputQueue persistenceDir capacity = do
+  q <- newPersistentQueue (persistenceDir </> "input-queue") capacity
+  pure
+    InputQueue
+      { enqueue = writePersistentQueue q
```

Comment:
Must: use a durable write here


In fact, we would not want durable writes on the persistent queue if this PR should solve the issue #2109. Otherwise we kicked the can only slightly down the road.